packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { Fragment, useCallback, useRef, SetStateAction } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  XXSmall: 'xxsmall',
+  XSmaill: 'xsmall',
+  Small: 'small',
+  Normal: 'normal',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+const overlayStyle = css`
+  animation: fade-in 500ms cubic-bezier(0.165, 0.84, 0.44, 1);
```

Comment:
Heh getting fancy with the timing functions! I'll get back to ya on this, but usually I don't like using the CSS `cubic-bezier` function because it only supports two points on the curve, where you need 3+ to make a bounce feel natural. May want to consider using a JS animation for this, but I'll get back to ya!

Reply:
Sounds good LMK